### PR TITLE
new JDBC connection configuration

### PR DESCRIPTION
**Pull request by mtbc**
Provides properties used by https://github.com/ome/omero-server/pull/108.

### CODE REVIEW

**Review by joshmoore — COMMENTED** [review]
Looks generally fine. Going back to https://jdbc.postgresql.org/documentation/head/connect.html I wonder a few things:

 - should "properties" be "options"?
 - should we add `&ApplicationName=OMERO` both for debugging connections but also to make the syntax clear?
 - should we link to the above URL? 
 - do we need to tell users that omero.db.url largely doesn't need changing?
 - does `tcpKeepAlive` negatively impact the freeing of resources from the connection pool?

**Comment by mtbc** [comment]
I avoided linking because then it's another embedded link that we're not actually checking in CI. Certainly could though. I'd stick with "properties" rather than "options" because that's what the linked page calls them. I suspect that we have many configuration settings that largely don't need changing! :smiley:

**Comment by joshmoore** [comment]
> I avoided linking because then it's another embedded link that we're not actually checking in CI. Certainly could though.

I'd rather have a URL that possibly is broken and let's someone know *roughly* what they're looking for, rather than no docs whatsoever. (Obviously, having a link checker would be great)

> I'd stick with "properties" rather than "options"

Hmm... I was thinking about this:

```
options = String

Specify 'options' connection initialization parameter.

The value of this property may contain spaces or other special characters, and it should be properly encoded if provided in the connection URL. Spaces are considered to separate command-line arguments, unless escaped with a backslash (\); \\ represents a literal backslash.
```
but I guess they are called "properties", "parameters", *and* "options" on that page.

File: src/main/resources/omero-model.properties [hunk 1]
```diff
@@ -86,6 +86,14 @@ omero.db.authority=export.openmicroscopy.org
 # max_threads is less than poolsize.
 omero.db.poolsize=10
 
+# Properties to set on OMERO.server's JDBC connection to the database.
+# Using TCP keepalive helps to prevent mysterious connection closures.
+# https://jdbc.postgresql.org/documentation/head/connect.html lists more.
+omero.db.properties=ApplicationName=OMERO&tcpKeepAlive=true
+
+# The URL specifying how the Java driver connects to the database system.
+omero.db.url=jdbc:postgresql://${omero.db.host}:${omero.db.port}/${omero.db.name}?${omero.db.properties}
```
Comment (sbesson) [inline src/main/resources/omero-model.properties:95]: While reviewing this property across our codebase, I found https://github.com/ome/openmicroscopy/blob/dd1584558cddf3079cceb16d05976d189b77116e/components/antlib/resources/lifecycle.xml#L116-L128. Should the latter be dropped or is is serving an additional purpose?

Comment (joshmoore) [reply src/main/resources/omero-model.properties:95]: `sql_run` is no longer being use (and can be dropped)

Comment (mtbc) [reply src/main/resources/omero-model.properties:95]: Thank you, opened https://github.com/ome/openmicroscopy/pull/6252 accordingly.

**Comment by joshmoore** [comment]
Note: tcpKeepAlive did _not_ solve the connection issue so holding off. Alternatively, we strip it and leave it configurable.... which I not means that if someone sets `properties` the application name will be _unset_.

**Comment by joshmoore** [comment]
> tcpKeepAlive did _not_ solve the connection issue so holding off

If we decide tcpKeepAlive is unsafe, I met suggest this as the default:

```
omero.db.properties=
omero.db.url=jdbc:postgresql://${omero.db.host}:${omero.db.port}/${omero.db.name}ApplicationName=OMERO&${omero.db.properties}
```

unless we have access to "Blitz", "Indexer", etc. in Spring properties.

**Comment by mtbc** [comment]
Aha, the default trailing `&` is happily ignored? Convenient.

**Comment by joshmoore** [comment]
> unless we have access to "Blitz", "Indexer", etc. in Spring properties.

Closest I see are:
```
Ice.Admin.ServerId=Indexer-0
Ice.ProgramName=Indexer-0
```

**Comment by mtbc** [comment]
Especially as we aim to retire it. :smiley: Happy to give it a try anyway if you like.

**Comment by joshmoore** [comment]
Ah! The command-line has: `-Domero.name=PixelData-0` so if we could define a default of "OMERO" then this should take precedence.

**Comment by joshmoore** [comment]
Thanks, @mtbc!

**Pull request merged**